src/shared/tpm2-util.c

```diff
@@ -1470,6 +1470,30 @@ int tpm2_digest_hash_digests(
                 (TPM2B_AUTH*)(digest);                                  \
         })
 
+static int tpm2_set_auth(Tpm2Context *c, const Tpm2Handle *handle, const char *pin) {
+        TPM2B_DIGEST digest;
+        TSS2_RC rc;
+        int r;
+
+        assert(c);
+
+        if (!pin)
+                return 0;
+
+        CLEANUP_ERASE(digest);
+
+        r = tpm2_digest_init_buffer(TPM2_ALG_SHA256, &digest, (uint8_t*) pin, strlen(pin));
```

Comment:
ditto (here and everywhere)

Reply:
ack, removed cast from here (but as noted still needed in tpm2_digest_buffer() define)